packages/server/src/base/api-manager.ts

```diff
@@ -33,3 +33,33 @@ export class ApiManager {
     )
   }
 }
+
+export class AdminApiManager {
+  constructor(private router: Router, private auth: Auth) {}
+
+  post(path: string, schema: Joi.ObjectSchema<any>, fn: Middleware<Request>) {
+    this.use('post', path, schema, fn)
+  }
+
+  get(path: string, schema: Joi.ObjectSchema<any>, fn: Middleware<Request>) {
+    this.use('get', path, schema, fn)
+  }
+
+  delete(path: string, schema: Joi.ObjectSchema<any>, fn: Middleware<Request>) {
+    this.use('delete', path, schema, fn)
+  }
+
+  use(type: 'post' | 'get' | 'delete', path: string, schema: Joi.ObjectSchema<any>, fn: Middleware<Request>) {
+    this.router[type](
+      path,
+      this.auth.public.auth(async (req, res) => {
```

Comment:
Shouldn't we rather use something like `this.auth.admin.auth` instead?

Reply:
Yup added that. There is now an ADMIN_KEY env variable that protects these routes